src/thing_objects.c

```diff
@@ -569,6 +569,18 @@ void destroy_object(struct Thing *thing)
     }
 }
 
+TbBool object_is_indestructable (const struct Thing* thing)
```

Comment:
You tell me. I was torn too, because I am conscious about avoiding double negatives too, still I chose this way as 'indestructible' is not a negative per se, and if I wrote 'object_is_destructible' there it seemed to imply it getting destroyed at this point.
Now the statement is 'Indestructible objects take no damage'.
Besides that, I'm specifically looking for an existing object, with the 'indestructible' property.

So, should I flip it you feel?

Reply:
to me it sounds like being able to take damage would be a property of something that also implies that it has health etc whereas a dumb item that does nothing would just sit there
but yeah could definitely see it both ways

Reply:
Well, that's complicated. In the code thing =! object. All objects are things, but there are non object things too. Things with negative health are destroyed, units, effects, shots, everything. But for objects the game expects a special function to destroy it in a pretty way, say, a dungeon heart will spark, a chicken egg will squish. The other objects at the moment do not know how to be destroyed and leave an error, but still get negative HP.

In short, objects specifically mostly cannot be destroyed, but most things can.

Reply:
only now notice it's under thing_objects and not under a generic file
shouldn't the function apply to all things and not a function under objects specifically?
creatures are also a seperate thing class and so not considered objects right, and you still use the function to damage creatures so thing_objects  might not be the best place
maybe can_take_damage would be an even better name then destructible
    
out of this list of thing classes the only ones I see would make sense to take damage would be
Creature, Door and maybe currently not so but trap and object could also make sense in some cases
but don't see how the other would take damage really so most can't

    TCls_Empty        =  0
    TCls_Object       =  1
    TCls_Shot         =  2,
    TCls_EffectElem   =  3,
    TCls_DeadCreature =  4,
    TCls_Creature     =  5,
    TCls_Effect       =  6,
    TCls_EffectGen    =  7,
    TCls_Trap         =  8,
    TCls_Door         =  9,
    TCls_Unkn10       = 10,
    TCls_Unkn11       = 11,
    TCls_AmbientSnd   = 12,
    TCls_CaveIn       = 13,

Reply:
only now notice it's under thing_objects and not under a generic file
shouldn't the function apply to all things and not a function under objects specifically?
creatures are also a separate thing class and so not considered objects right, and you still use the function to damage creatures so thing_objects might not be the best place
maybe can_take_damage would be an even better name then destructible

out of this list of thing classes the only ones I see can be damaged
Creature, Door and some objects
3/13 classes so that's clearly not most